src/Compilers/CSharp/Portable/Compiler/ClsComplianceChecker.cs

```diff
@@ -39,6 +43,11 @@ internal partial class ClsComplianceChecker : CSharpSymbolVisitor
             _cancellationToken = cancellationToken;
 
             _declaredOrInheritedCompliance = new ConcurrentDictionary<Symbol, Compliance>();
+
+            if (compilation.Options.ConcurrentBuild)
```

Comment:
>compilation.Options.ConcurrentBuild [](start = 16, length = 35)

Consider also checking if _filterTree is null.

Reply:
➡️ I thought about it, but figured it risked regression if the conditions ever changed. The one thing we can count on is it's only needed during concurrent builds.

Reply:
>  let me know and I'll extract the condition to a property

This feels like the right thing to do. Why allocate when we don't need to? 

---
In reply to: [165447222](https://github.com/dotnet/roslyn/pull/23519#discussion_r165447222) [](ancestors = 165447222)